music_assistant/providers/roku_media_assistant/player.py

```diff
@@ -0,0 +1,386 @@
+        """Handle enqueuing of the next (queue) item on the player."""
+        try:
+            device_info = await self.roku.update()
+
+            app_running = False
+
+            if device_info.app is not None:
+                app_running = device_info.app.app_id == self.provider.config.get_value(
+                    CONF_ROKU_APP_ID
+                )
+
+            if app_running:
+                await self.roku_input(
+                    {
+                        "u": media.uri,
+                        "t": "a",
+                        "albumName": media.album,
+                        "songName": media.title,
+                        "artistName": media.artist,
+                        "albumArt": media.image_url,
+                        "songFormat": "flac",
+                        "duration": media.duration,
+                        "enqueue": "true",
+                    },
+                )
+                self.queued = media
+        except Exception:
+            self.logger.error("Failed to Enqueue Media on: %s", self.name)
+            return
+
+    async def poll(self) -> None:
+        """Poll player for state updates."""
+        # Pull Device State
+        try:
+            device_info = await self.roku.update()
+            self._attr_available = True
+        except Exception:
+            self._attr_available = False
+            self.logger.error("Failed to retrieve Update from: %s", self.name)
+            self.update_state()
+            return
+
+        app_running = False
+
+        if device_info.app is not None:
+            app_running = device_info.app.app_id == self.provider.config.get_value(CONF_ROKU_APP_ID)
+
+        # Update Device State
+        if not app_running:
+            self._attr_active_source = None
+
+        self._attr_powered = app_running
+
+        # If Media's Playing update its state
+        if self.powered and app_running:
+            try:
+                media_state = await self.roku._get_media_state()
+
+                play_states: dict[str, PlaybackState] = {
+                    "play": PlaybackState.PLAYING,
+                    "pause": PlaybackState.PAUSED,
+                }
+
+                self._attr_playback_state = play_states.get(
+                    media_state["@state"], PlaybackState.IDLE
+                )
+
+                if "position" in media_state:
+                    try:
+                        self._attr_elapsed_time = (
+                            int(media_state["position"].split(" ", 1)[0]) / 1000
+                        )
+                        self._attr_elapsed_time_last_updated = time.time()
+                    except Exception:
+                        self.logger.info(
+                            "Playback Position received from %s Was Invalid", self.name
+                        )
+
+                if not (queue := self.mass.player_queues.get_active_queue(self.player_id)):
```

Comment:
```suggestion
                if not (queue := self.mass.player_queues.get_active_queue(self.current_media.source_id)):
```
See my comment above about the source id on the media